src/middlewared/middlewared/plugins/system.py

```diff
@@ -1359,11 +1401,35 @@ def set_crash_reporting(self):
         CrashReporting.enabled_in_settings = self.middleware.call_sync('system.general.config')['crash_reporting']
 
 
+# Update Birthday Date
+async def _update_birthday(middleware):
+    # Sync clock
+    middleware.logger.debug('Synchornization the clock for system birthday')
+    birthday = None
+    timeout = 60
+
+    while birthday is None:
+        birthday = await middleware.call('system.sync_clock')
+        # Wait until be able to sync the clock
+        if birthday is None:
+            await asyncio.sleep(timeout)
+
+    settings = middleware.call_sync('datastore.config', 'system.settings')
+    middleware.call_sync('datastore.update', 'system.settings', settings['id'], {
+        'stg_birthday': birthday,
+    })
+
+
 async def _event_system(middleware, event_type, args):
     global SYSTEM_READY
     global SYSTEM_SHUTTING_DOWN
     if args['id'] == 'ready':
         SYSTEM_READY = True
+        # Check if birthday is already setted
+        birthday = middleware.sync_call('system.info')['birthday']
```

Comment:
Sorry, but I am under impression that this code has never been executed before pushing

Reply:
Simulated now with: `midclt call core.event_send system ADDED '{"id": "ready"}'`